amaranth_boards/tang_nano_9k.py

```diff
@@ -0,0 +1,135 @@
+import os
+import subprocess
+
+from amaranth.build import *
+from amaranth.vendor.gowin import *
+
+try:
+    from .resources import *
+except:
+    from resources import *
+
+
+__all__ = ["TangNano9kPlatform"]
+
+
+class TangNano9kPlatform(GowinPlatform):
+    part          = "GW1NR-LV9QN88PC6/I5"
+    family        = "GW1NR-9C"
+    default_clk   = "clk27" # or "OSC" to use on-chip oscillator
+    osc_frequency = 2500000 # Hz
+    board         = "tangnano9k"
```

Comment:
Our convention is to inline the `board` field you have here into `toolchain_program`.